src/TreeContainer.js

```diff
@@ -145,7 +184,16 @@ class TreeContainer extends Component {
     }
 
     shouldComponentUpdate(nextProps) {
-        return nextProps._dashprivate_layout !== this.props._dashprivate_layout;
+        const { _dashprivate_layout, _dashprivate_requestQueue } = nextProps;
+
+        return _dashprivate_layout !== this.props._dashprivate_layout ||
+            (this.isDeepStateElement() && _dashprivate_requestQueue !== this.props._dashprivate_requestQueue);
```

Comment:
non-deep components only update on layout changes,
deep components update on layout changes and request queue changes